index.js

```diff
@@ -21,7 +23,12 @@ var s3urls = module.exports = {
       key = uri.pathname.split('/').slice(2).join('/');
     }
     if (style === 'bucket-in-host') {
-      bucket = uri.hostname.split('.')[0];
+      const match = uri.hostname.match(new RegExp(/.*?(?=.s3)/));
+      if (match) {
+        bucket = match[0];
+      } else {
+        bucket =  uri.hostname.split('.')[0];
+      }
```

Comment:
@ingalls am I right that the above is the only substantive change in the library (plus your additional tests) ?

Reply:
Yes

Reply:
Can you add a testcase for `results.s3llout-to-the-man.io.s3.amazonaws.com`

Reply:
good :eyes: , done